File: maps/src/main/java/it/localhostsoftware/maps/google/model/GoogleBitmapDescriptorFactory.java
package it.localhostsoftware.maps.google.model;

import android.graphics.Bitmap;

import it.localhostsoftware.maps.model.BitmapDescriptor;
import it.localhostsoftware.maps.model.BitmapDescriptorFactory;

public class GoogleBitmapDescriptorFactory implements BitmapDescriptorFactory {
    @Override
    public BitmapDescriptor<?> fromResource(int var0) {
        return new GoogleBitmapDescriptor(com.google.android.gms.maps.model.BitmapDescriptorFactory.fromResource(var0));
    }

    @Override
    public BitmapDescriptor<?> fromAsset(String var0) {
        return new GoogleBitmapDescriptor(com.google.android.gms.maps.model.BitmapDescriptorFactory.fromAsset(var0));
    }

    @Override
    public BitmapDescriptor<?> fromFile(String var0) {
        return new GoogleBitmapDescriptor(com.google.android.gms.maps.model.BitmapDescriptorFactory.fromFile(var0));
    }

    @Override
    public BitmapDescriptor<?> fromPath(String var0) {
        return new GoogleBitmapDescriptor(com.google.android.gms.maps.model.BitmapDescriptorFactory.fromPath(var0));
    }

    @Override
    public BitmapDescriptor<?> defaultMarker() {
        return new GoogleBitmapDescriptor(com.google.android.gms.maps.model.BitmapDescriptorFactory.defaultMarker());
    }

    @Override
    public BitmapDescriptor<?> defaultMarker(float var0) {
        return new GoogleBitmapDescriptor(com.google.android.gms.maps.model.BitmapDescriptorFactory.defaultMarker(var0));
    }

    @Override
    public BitmapDescriptor<?> fromBitmap(Bitmap var0) {
        return new GoogleBitmapDescriptor(com.google.android.gms.maps.model.BitmapDescriptorFactory.fromBitmap(var0));
    }
}

File: maps/src/main/java/it/localhostsoftware/maps/google/model/GoogleBitmapDescriptorFactory.kt
package it.localhostsoftware.maps.google.model

import android.graphics.Bitmap
import it.localhostsoftware.maps.model.BitmapDescriptor
import it.localhostsoftware.maps.model.BitmapDescriptorFactory

class GoogleBitmapDescriptorFactory : BitmapDescriptorFactory {
    override fun fromResource(var0: Int) =
            BitmapDescriptor(com.google.android.gms.maps.model.BitmapDescriptorFactory.fromResource(var0))

    override fun fromAsset(var0: String) =
            BitmapDescriptor(com.google.android.gms.maps.model.BitmapDescriptorFactory.fromAsset(var0))

    override fun fromFile(var0: String) =
            BitmapDescriptor(com.google.android.gms.maps.model.BitmapDescriptorFactory.fromFile(var0))

    override fun fromPath(var0: String) =
            BitmapDescriptor(com.google.android.gms.maps.model.BitmapDescriptorFactory.fromPath(var0))

    override fun defaultMarker() =
            BitmapDescriptor(com.google.android.gms.maps.model.BitmapDescriptorFactory.defaultMarker())

    override fun defaultMarker(var0: Float) =
            BitmapDescriptor(com.google.android.gms.maps.model.BitmapDescriptorFactory.defaultMarker(var0))

    override fun fromBitmap(var0: Bitmap) =
            BitmapDescriptor(com.google.android.gms.maps.model.BitmapDescriptorFactory.fromBitmap(var0))
}